features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/telemetry/internal/TelemetryEventHandler.kt

```diff
@@ -322,12 +332,57 @@ internal class TelemetryEventHandler(
                     touchPrivacyLevel = sessionReplayTouchPrivacy,
                     textAndInputPrivacyLevel = sessionReplayTextAndInputPrivacy,
                     startRecordingImmediately = startRecordingImmediately,
-                    batchProcessingLevel = coreConfiguration.batchProcessingLevel.toLong()
+                    batchProcessingLevel = event.batchProcessingLevel.toLong()
                 )
             )
         )
     }
 
+    private fun createApiUsageEvent(
+        datadogContext: DatadogContext,
+        timestamp: Long,
+        event: InternalTelemetryEvent.ApiUsage
+    ): TelemetryUsageEvent? {
+        val rumContext = datadogContext.rumContext()
+        return when (event) {
+            is InternalTelemetryEvent.ApiUsage.AddViewLoadingTime -> {
+                TelemetryUsageEvent(
+                    dd = TelemetryUsageEvent.Dd(),
+                    date = timestamp,
+                    source = TelemetryUsageEvent.Source.tryFromSource(
+                        datadogContext.source,
+                        sdkCore.internalLogger
+                    ) ?: TelemetryUsageEvent.Source.ANDROID,
+                    service = datadogContext.service,
+                    version = datadogContext.sdkVersion,
+                    application = TelemetryUsageEvent.Application(rumContext.applicationId),
+                    session = TelemetryUsageEvent.Session(rumContext.sessionId),
+                    view = rumContext.viewId?.let { TelemetryUsageEvent.View(it) },
+                    action = rumContext.actionId?.let { TelemetryUsageEvent.Action(it) },
+                    telemetry = TelemetryUsageEvent.Telemetry(
+                        additionalProperties = event.additionalProperties,
+                        device = TelemetryUsageEvent.Device(
+                            architecture = datadogContext.deviceInfo.architecture,
+                            brand = datadogContext.deviceInfo.deviceBrand,
+                            model = datadogContext.deviceInfo.deviceModel
+                        ),
+                        os = TelemetryUsageEvent.Os(
+                            build = datadogContext.deviceInfo.deviceBuildId,
+                            version = datadogContext.deviceInfo.osVersion,
+                            name = datadogContext.deviceInfo.osName
+                        ),
+                        usage = TelemetryUsageEvent.Usage.AddViewLoadingTime(
+                            overwritten = event.overwrite,
+                            noView = event.noView,
+                            noActiveView = event.noActiveView
+                        )
+                    )
+                )
+            }
+            else -> null
```

Comment:
we don't need `else` here I think, because we are using sealed class. So if we remove `else` here and add one more sub-type to `InternalTelemetryEvent.ApiUsage`, without added additional branch in `when`, we will get a compilation error, which is for good.

Reply:
yup it is exactly the case in line `117` also so you are right